### PR TITLE
Build the images as a base image, instead of installing all the time

### LINKED ISSUE #37: Build the images as a base image, instead of installing all the time
Everytime a customer deploys a preview env, github builds a docker image for our action. Our image takes close to a minute to build, because we do the following:
1. install ruby dependencies
2. install jq
3. build a go binary
4. copy the okteto binary

I think we could create a base image that can be used by all of our images with those 4 things, so that we don't have to do that on every preview env. Or at the very least the first 3. That reduces our dependency on 3dr party systems (e.g. rubygems being online) and should result on faster deploys (in my experience, it went from 40 seconds to 10 seconds to build the action container).

### PR DESCRIPTION
**Pull request by rberrelleza**
Fixes #37 

This is an experiment to see if using a base image with ruby, jq, and the go binary prebuilt makes the deploy-preview action faster.  

PROS:
- In my tests, action build time went from 40 seconds to 10 seconds. 
- We are not installing JQ, faraday, and building the go binary all the time

CONS:
- we might need to pay for storage and distribution of the image (we can mitigate this by moving it to dockerhub)
- the action repo now has two different Dockerfiles and  there could be races when modifying both the message.go file and the action
- the image is displayed in the 'package'  part of the repo.

If we go with this approach, I think we should move the dependencies to a different repository, and leave only the action code (action.yml, entrypoint.sh) here. 

- [x] update the action to only trigger on main merge

### CODE REVIEW

**Review by pchico83 — APPROVED** [review]

**Review by ifbyol — APPROVED** [review]

**Review by jLopezbarb — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -1,21 +1,10 @@
...
-RUN go get github.com/machinebox/graphql
-COPY message.go .
-RUN go build -o /message .
-RUN curl https://stedolan.github.io/jq/download/linux64/jq > /usr/bin/jq && chmod +x /usr/bin/jq
 
-FROM ruby:3-slim-buster
-
-RUN gem install octokit
+FROM okteto/okteto:2.0.1 as okteto
+FROM ghcr.io/okteto/deploy-preview:latest
```
Comment (rberrelleza) [inline Dockerfile:3]: @pchico83 @jLopezbarb using the latest there's a risk that the base image is not compatible with the image running on the tag. would it be ok if we create the image when creating the release instead?

Comment (rberrelleza) [reply Dockerfile:3]: another option (my favorite) is to move the base image to another repo

Comment (pchico83) [reply Dockerfile:3]: +1 for moving the base image to a different repo.
@rberrelleza if you want, create an issue for it, and we will take care of adding it to the planning

Comment (rberrelleza) [reply Dockerfile]: https://github.com/okteto/actions-base has the base dependencies now.

**Comment by rberrelleza** [comment]
@pchico83 we need to publish the 1.0 version on https://github.com/okteto/actions-base before merging this.   Waiting on permissions so I can trigger it.

**Pull request closed without merging**